gremlin-test/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/map/VertexTest.java

```diff
@@ -276,6 +277,7 @@ public void g_EXe7_e11X() {
     }
 
     @Test
+    @Ignore("g.E(Arrays.asList(e7, e11)) is not valid for gremlin-lang")
```

Comment:
ignoring at this level will ignore for embedded tests. the change should be to ignore for gremlin-lang only, right?

Reply:
This is not valid for Grammar, probably need to clean up this a bit.

in Grammar we allow `g.V(vertex)` and `g.E(vertex)`, but not `g.V(edge)` and `g.E(edge)`.
gremlin-lang and gremlin-groovy can handle `g.V(vertex)` and `g.E(edge)`.

For gremlin-lang `g.E(edge)` translated as `g.E(_1)` because no Edge defined in Grammar. But this approach not valid for `g.E([edge1, edge2])`, gremlin-lang can't handle Parameters inside array.

Reply:
> in Grammar we allow g.V(vertex) and g.E(vertex), but not g.V(edge) and g.E(edge).

what is `g.E(vertex)` and `g.V(edge)`? 

> For gremlin-lang g.E(edge) translated as g.E(_1) because no Edge defined in Grammar.

i think think that perhaps the problem here is that the grammar was adapted to allow the tests to work. from a grammar perspective, i'm not sure that anyone needs to do `g.V(Vertex(1,'person'))` (or the edge equivalent). You would simply do `g.V(1)`. i think `g.V(Vertex)` though is useful at the API level because you can do:

```groovy
v = g.V(1).next()
g.addE('self').from(v).to(v)
```

but again, that's from a programming language perspective not the query perspective. might be smart to take something away here. 

> But this approach not valid for g.E([edge1, edge2]), gremlin-lang can't handle Parameters inside array.

i think i know why we allowed that `List` syntax:

```groovy
l = g.V(1,2,3).toList()
g.V(l)
```

i wonder if that's a good idea though. maybe we're doing too much with this sort of overload.

Reply:
> > in Grammar we allow g.V(vertex) and g.E(vertex), but not g.V(edge) and g.E(edge).
> 
> what is `g.E(vertex)` and `g.V(edge)`?
> 
`vertex` is some Vertex object, so `g.E(Vertex(1,'person'))` is valid for Grammar

> 
> i wonder if that's a good idea though. maybe we're doing too much with this sort of overload.

I assume we can limit possible arguments for `V()` step aka traversalSourceSpawnMethod_V, because now there are too many options that make no sense, for example `g.V(Merge.outV)` and `g.V(datetime(...))` is kind of valid for Grammar.
Same for `E()` step